packages/esm-patient-immunizations-app/src/hooks/useImmunizationsConceptSet.tsx

```diff
@@ -0,0 +1,17 @@
+import { openmrsFetch } from '@openmrs/esm-framework';
+import useSWRImmutable from 'swr/immutable';
+import { type ImmunizationWidgetConfigObject, type OpenmrsConcept } from '../types/fhir-immunization-domain';
+
+export function useImmunizationsConceptSet(config: ImmunizationWidgetConfigObject) {
+  const conceptRepresentation =
+    'custom:(uuid,display,answers:(uuid,display),conceptMappings:(conceptReferenceTerm:(conceptSource:(name),code)))';
+
+  const { data, error, isLoading } = useSWRImmutable<{ data: { results: Array<OpenmrsConcept> } }, Error>(
+    `/ws/rest/v1/concept?references=${config.vaccinesConceptSet}&v=${conceptRepresentation}`,
```

Comment:
What is the `vaccineConceptSet`? I find this a little confusing because immunizations themselves are defined via a concept set.

Reply:
The `vaccineConceptSet` is a configuration property that serves as a reference to the identifier of the concept set associated with immunizations. I understand the confusion – the naming of this property is not intuitive enough. 

Gonna rename it to: `immunizationConceptSet`